nvflare/app_common/storages/filesystem_storage.py

```diff
@@ -62,6 +62,21 @@ def _object_exists(uri: str):
     return all((os.path.isabs(uri), os.path.isdir(uri), data_exists, meta_exists))
 
 
+def _encode_meta(meta: dict) -> bytes:
+    return json.dumps(meta).encode("utf-8")
+
+
+def _decode_meta(data: bytes) -> dict:
+    s = data.decode("utf-8")
+    if s.startswith('"'):
+        # this is in old format
+        result = ast.literal_eval(json.loads(s))
```

Comment:
not about the logic, 
I am just thinking, if there is any possibility some people can put weird stuff in the meta?
But I guess that means they already compromise the file system on server side, 
we have nothing we can do by that time then

Reply:
No I don't think we need to worry about that. The problem is that our system may put some strings in the data that literal_eval doesn't like. Remember that we keep a lot of info about the job in the meta, and we may put even more there. A lot of info is free strings.